src/base/bittorrent/private/bandwidthscheduler.cpp

```diff
@@ -24,72 +25,78 @@
  * modify file(s), you may extend this exception to your version of the file(s),
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
- *
- * Contact : chris@qbittorrent.org
  */
 
-#include <QTime>
+#include "bandwidthscheduler.h"
+
+#include <utility>
+
 #include <QDateTime>
+#include <QTime>
 
-#include "base/bittorrent/session.h"
 #include "base/preferences.h"
-#include "bandwidthscheduler.h"
 
 BandwidthScheduler::BandwidthScheduler(QObject *parent)
-    : QTimer(parent)
+    : QObject(parent)
+    , m_lastAlternative(false)
 {
-    // Single shot, we call start() again manually
-    setSingleShot(true);
-    // Connect Signals/Slots
-    connect(this, SIGNAL(timeout()), this, SLOT(start()));
+    connect(&m_timer, &QTimer::timeout, this, &BandwidthScheduler::onTimeout);
 }
 
 void BandwidthScheduler::start()
 {
+    m_lastAlternative = isTimeForAlternative();
+    emit bandwidthLimitRequested(m_lastAlternative);
+
+    // Timeout regularly to accommodate for external system clock changes
+    // eg from the user or from a timesync utility
+    m_timer.start(30000);
+}
+
+bool BandwidthScheduler::isTimeForAlternative()
```

Comment:
It doesn't matter in this case because there is no one `const` public method. But I still fixed it along the way.